tools/fuzzer/fuzzer_test.cc

```diff
@@ -545,6 +545,46 @@ std::vector<PrecedenceTestParam> gen_precedence_params() {
     std::string str = "-1.5";
     params.emplace_back(std::move(str), std::move(expected));
   }
+  {
+    Expr expected = IntegerConstant(0xbadf00d, IntegerConstant::Base::Hex,
+                                    IntegerConstant::Length::LongLong,
+                                    IntegerConstant::Signedness::Unsigned);
+
+    std::string str = "0xbadf00dLLU";
+    params.emplace_back(std::move(str), std::move(expected));
+  }
+  {
+    Expr expected = IntegerConstant(0b1001'1001, IntegerConstant::Base::Bin,
+                                    IntegerConstant::Length::Long,
+                                    IntegerConstant::Signedness::Signed);
+
+    std::string str = "0b10011001L";
+    params.emplace_back(std::move(str), std::move(expected));
+  }
+  {
+    Expr expected = DoubleConstant(1.5, DoubleConstant::Format::Default,
+                                   DoubleConstant::Length::Float);
+
+    std::string str = "1.5f";
+    params.emplace_back(std::move(str), std::move(expected));
+  }
+  {
+    Expr expected = DoubleConstant(0x0.1p-1070, DoubleConstant::Format::Hex,
+                                   DoubleConstant::Length::Double);
+
+#ifdef _WIN32
+    // Apparently MSVC seems to not be adhering to the C++11 standard properly
+    // and it takes into account the precision modifier when printing hex
+    // floats.
+    //
+    // Hence we're using this ifdef guard as a temporary workaround.
+    std::string str = "0x0.000000p-1022";
+#else
+    std::string str = "0x0.0000000000001p-1022";
+#endif
```

Comment:
@werat I personally don't like this approach for working around the printing issue on Windows. Any suggestions?